pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/MockedPulsarServiceBaseTest.java

```diff
@@ -110,7 +111,11 @@ protected final void internalSetup() throws Exception {
         if (isTcpLookup) {
             lookupUrl = new URI("pulsar://localhost:" + BROKER_PORT);
         }
-        pulsarClient = PulsarClient.builder().serviceUrl(lookupUrl.toString()).statsInterval(0, TimeUnit.SECONDS).build();
+        setPulsarClient(lookupUrl.toString(), 0);
+    }
+
+    protected void setPulsarClient(String url, int intervalInSecs) throws PulsarClientException {
```

Comment:
This method can be called many times in a single tests, and each time a new `pulsarclient` is created, but they are not explicitly closed. While in the original logic, once test completed it will call `client.close()`. From my view it would be better to return each new created pulsarClient, and close the one that additionally added.

Reply:
Sorry, what I said was incorrect because would result on previous client connection being close, better to be returned as a result of `setPulsarClient` ?

Reply:
@jiazhai Could you please take a look now? I did some changes according to your suggestions I think, Thanks!